src/LargeXlsx/Util.cs

```diff
@@ -28,107 +28,112 @@ OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN
-            {
-                columnName = GetColumnNameInternal(columnIndex);
-                CachedColumnNames[columnIndex - 1] = columnName;
-            }
-            return columnName;
-        }
-
-        private static string GetColumnNameInternal(int columnIndex)
-        {
-            var columnName = new StringBuilder(3); // This has been measured to be faster than string concatenation
-            while (true)
-            {
-                if (columnIndex > 26)
-                {
-                    columnIndex = Math.DivRem(columnIndex - 1, 26, out var rem);
-                    columnName.Insert(0, (char)('A' + rem));
-                }
-                else
-                {
-                    columnName.Insert(0, (char)('A' + columnIndex - 1));
-                    return columnName.ToString();
-                }
-            }
-        }
-
-        public static double DateToDouble(DateTime date)
-        {
-            var days = date.Subtract(ExcelEpoch).TotalDays + 1;
-            // Excel wrongly assumes that 1900 is a leap year:
-            // https://docs.microsoft.com/en-us/office/troubleshoot/excel/wrongly-assumes-1900-is-leap-year
-            if (date >= Date19000301) days++;
-            return days;
-        }
-
-        public static int BoolToInt(bool value) => value ? 1 : 0;
-
-        public static string EnumToAttributeValue<T>(T enumValue)
-        {
-            var s = enumValue.ToString();
-            return char.ToLowerInvariant(s[0]) + s.Substring(1);
-        }
-
-        // Hashing procedure courtesy of https://docs.microsoft.com/en-us/openspecs/office_file_formats/ms-offcrypto/1357ea58-646e-4483-92ef-95d718079d6f
-        public static byte[] ComputePasswordHash(string password, byte[] saltValue, int spinCount)
-        {
-            var hasher = new SHA512Managed();
-            var hash = hasher.ComputeHash(saltValue.Concat(Encoding.Unicode.GetBytes(password)).ToArray());
-            for (var i = 0; i < spinCount; i++)
-            {
-                var iterator = BitConverter.GetBytes(i);
-                if (!BitConverter.IsLittleEndian)
-                    Array.Reverse(iterator);
-                hash = hasher.ComputeHash(hash.Concat(iterator).ToArray());
-            }
-            return hash;
-        }
-    }
+	internal static class Util
+	{
+		private const int MaxColumnCount = 16384;
+		private static readonly DateTime ExcelEpoch = new DateTime(1900, 1, 1);
+		private static readonly DateTime Date19000301 = new DateTime(1900, 3, 1);
+		private static readonly string[] CachedColumnNames = new string[MaxColumnCount];
+
+		public static string EscapeXmlText(string value)
+		{
+			return System.Security.SecurityElement.Escape(RemoveInvalidXmlChars(value));
+		}
+
+		public static string EscapeXmlAttribute(string value)
+		{
+			return System.Security.SecurityElement.Escape(RemoveInvalidXmlChars(value));
+		}
+
+		public static string RemoveInvalidXmlChars(string value)
```

Comment:
I have mixed feelings about this function. If I understand it's purpose correctly, what are the differences between this function and XmlConvert.VerifyXmlChars? Moreover, I'm not 100% sure validating UTF-16 input should be responsibility of LargeXlsx, also introducing a performance hit. My usual reasoning is that you should ensure UTF representations are valid in your code, before passing strings to any library, but I don't have strong opinions to be honest.

Reply:
The problem is that a string value that is passed to a cell can have special characters like (char)2 (start of text) that will brake the resulting xlsx because we output an invalid xml. This was the modification needed for me to export xlsx from a database correctly.
You can try it yourself with the current release version in writing a string cell value which contains a (char)2 and you will see what I mean.